### PR TITLE
fix: set rule index result by security rules mapping

### PR DESCRIPTION
**Pull request by ShaharAvigezer**
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/CONTRIBUTING.md) rules

#### What does this PR do?
Set the `ruleIndex` of a result to match the index of the rule in the security rules map

#### Where should the reviewer start?
`src/lib/plugins/sast/analysis.ts`

#### How should this be manually tested?
run `code test` on a repo, in each result, look for rule index and make sure it matches in the rules array

#### Any background context you want to provide?
When we run a code test, we first alter the rules array to use only security rules. The source of the problem lies when we don't address the `ruleIndex` that is now outdate since the rules indexes have changed.

#### What are the relevant tickets?
[Jira Ticket](https://snyksec.atlassian.net/browse/ZEN-23)

#### Screenshots
N/A

#### Additional comments
I've also fixed the test fixture to match the desired result

### CODE REVIEW

**Review by Spoor2709 — CHANGES_REQUESTED** [review]
Required: Looks good to me can you put the debug change into a different commit please 😃

File: .vscode/launch.json [hunk 1]
```diff
@@ -22,6 +22,19 @@
       "console": "integratedTerminal",
       "program": "${workspaceFolder}/node_modules/.bin/jest",
       "args": ["${relativeFile}"]
+    },
+    {
+      "type": "node",
+      "request": "launch",
+      "name": "Debug Jest Current File",
```
Comment (JackuB) [inline .vscode/launch.json:29]: Why is this needed? 🤔 I think code breakpoints/debugger works with the Jest launcher above as well?

And do you need `runInBand` (serial run) when you are running only a single file?

Comment (ShaharAvigezer) [reply .vscode/launch.json:29]: The existing launcher didn't work for me so I added this enhancement. It's fine by me to drop it if you don't believe it's necessary. WDYT?

**Review by Spoor2709 — APPROVED** [review]
Looks good to me, Great Job!

**Pull request merged**